picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -821,14 +831,16 @@ public Builder loggingEnabled(boolean enabled) {
     public Picasso build() {
       Context context = this.context;
 
-      if (downloader == null) {
+      okhttp3.Cache unsharedCache;
+      if (callFactory == null) {
         File cacheDir = createDefaultCacheDir(context);
         long maxSize = calculateDiskCacheSize(cacheDir);
-        okhttp3.Cache cache = new okhttp3.Cache(cacheDir, maxSize);
-        OkHttpClient client = new OkHttpClient.Builder()
-            .cache(cache)
+        unsharedCache = new okhttp3.Cache(cacheDir, maxSize);
+        callFactory = new OkHttpClient.Builder()
+            .cache(unsharedCache)
             .build();
-        downloader = new OkHttp3Downloader(client, cache, false);
+      } else {
+        unsharedCache = null;
```

Comment:
Initialize to `null` and then no `else` is needed.

Reply:
I'd just inline `unsharedCache` into the builder.